File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */

var twoSum = function(nums, target){
    const prevMap = new Map();

    for (i = 0; i < nums.length; i++){
        const n = nums[i];
        let diff = target - n;

        if (prevMap.has(diff)){
            return [prevMap.get(diff),i]; //[] to create an array with index of the diff
        }
    prevMap.set(n,i); //neu ma diff chua xuat hien trong hash ma thi set de bo vo hashmap
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        #map. key: num in array, val: the index of that number 
        map = {}
        for i, num in enumerate(nums):
            complement = target - num
            if complement in map:
                return [i, map[complement]]
            else:
                map[num] = i
        return None